holoviews/core/util.py

```diff
@@ -1532,7 +1531,7 @@ def groupby_python(self_or_cls, ndmapping, dimensions, container_type,
         selects = get_unique_keys(ndmapping, dimensions)
         selects = group_select(list(selects))
         groups = [(k, group_type((v.reindex(idims) if hasattr(v, 'kdims')
-                                  else [((), (v,))]), **kwargs))
+                                  else [((), v)]), **kwargs))
```

Comment:
These are NdMapping implementation details left over from when we had an NdElement, the precursor to datasets, which now lead to strange behavior.